#! /usr/bin/env python
#Autogenerated V4L Dynamic Control

PACKAGE='tuw_uvc'
from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

gen.add("show_camera_image", bool_t, 0, "Show camera image", True)
gen.add("camera_freeze", bool_t, 0, "Pulbishes the last image over and over again", True)
enum_convert_image = gen.enum([gen.const("raw_data", int_t, 0, "raw camera Data"),  gen.const("YUV422toRGB", int_t, 1, "converts image to rgb first"),  gen.const("YUV422toBRG", int_t, 2, "converts image to bgr first"),  gen.const("YUV422toGray", int_t, 3, "converts image to gray first")], "Convert image")
gen.add("convert_image_first", int_t, 3, "Convets the raw image first to an other format", 1, 0, 3, edit_method=enum_convert_image)

#Autogenerated Controls

gen.add("brightness", int_t, 0, "Brightness", 8, 0, 15)
gen.add("contrast", int_t, 0, "Contrast", 8, 0, 15)
gen.add("saturation", int_t, 0, "Saturation", 7, 0, 15)
gen.add("hue", int_t, 0, "Hue", 0, -10, 10)
gen.add("white_balance_temperature_auto", bool_t, 0, "White Balance Temperature, Auto", True)
gen.add("gamma", int_t, 0, "Gamma", 7, 1, 10)
gen.add("gain", int_t, 0, "Gain", 0, 0, 0)
enum_power_line_frequency = gen.enum([gen.const("Disabled", int_t, 0, "Disabled"), gen.const("50_Hz", int_t, 1, "50 Hz"), gen.const("60_Hz", int_t, 2, "60 Hz")], "Power Line Frequency")
gen.add("power_line_frequency", int_t, 0, "Power Line Frequency", 2, 0, 2, edit_method=enum_power_line_frequency)
gen.add("white_balance_temperature", int_t, 0, "White Balance Temperature", 2800, 2800, 6500)
gen.add("sharpness", int_t, 0, "Sharpness", 6, 0, 15)
gen.add("backlight_compensation", int_t, 0, "Backlight Compensation", 0, 0, 1)
enum_exposure_auto = gen.enum([gen.const("Unkown0", int_t, 0, "Unkown"), gen.const("Manual_Mode", int_t, 1, "Manual Mode"), gen.const("Unkown2", int_t, 2, "Unkown"), gen.const("Aperture_Priority_Mode", int_t, 3, "Aperture Priority Mode")], "Exposure, Auto")
gen.add("exposure_auto", int_t, 0, "Exposure, Auto", 3, 0, 3, edit_method=enum_exposure_auto)
gen.add("exposure_absolute", int_t, 0, "Exposure (Absolute)", 625, 4, 5000)
gen.add("focus_absolute", int_t, 0, "Focus (absolute)", 16, 0, 21)
gen.add("focus_auto", bool_t, 0, "Focus, Auto", True)

exit(gen.generate(PACKAGE, "tuw_uvc", "CameraWideAngle"))
